frappe/model/document.py

```diff
@@ -646,6 +646,9 @@ def get_value_before_save(self, fieldname):
 	def set_new_name(self, force=False, set_name=None, set_child_names=True):
 		"""Calls `frappe.naming.set_new_name` for parent and child docs."""
 
+		if self.name:
+			self.flags.name_set = True
```

Comment:
This does feel weird given the rest of the code, kind of has potentially to break some other behaviour. If we wanted to not override name we would've just done so. 

E.g. amending? desk's temporary `new-xyzx...` names?

Reply:
Desk is working because we are erasing temporary name.

https://github.com/frappe/frappe/blob/6e7cc59613cd5037e7060b58b0eb80114a1dcdb9/frappe/desk/form/save.py#L90-L94

Reply:
Well ig we can assume that those who are doing these temporary names should expect it to get saved? 

But then again the original issue this PR is fixing was also not advertised behavior.

https://www.hyrumslaw.com/ :laughing: